packages/common/src/HardhatConfig.js

```diff
@@ -67,6 +73,28 @@ function getHardhatConfig(configOverrides, workingDir = "./") {
         url: getNodeUrl("mainnet", true),
         accounts: { mnemonic },
       },
+      optimism: {
+        url: "http://127.0.0.1:8545",
+        accounts: { mnemonic },
+        // This sets the gas price to 0 for all transactions on L2. We do this because account balances are not yet
+        // automatically initiated with an ETH balance.
+        gasPrice: 0,
+        // This sets the network as using the ovm and ensure contract will be compiled against that.
+        ovm: true,
+        ignoreRxList: [
```

Comment:
I ignore every single directory right now because the Optimism-solc fork does not work exactly the same as solc. Certain Solidity code cannot be compiled so easily, such as inline assembly. Additionally, the logs do not make it clear which contracts are failing to compile, so for demonstration purposes I ignore every directory except the `optimism` one.

Here's an example compile error:

```
yarn hardhat compile --network optimism --show-stack-traces
yarn run v1.22.10
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/hardhat compile --network optimism --show-stack-traces

Compiling 191 files with 0.8.4
OVM Compiler Error (insert "// @unsupported: ovm" if you don't want this file to be compiled for the OVM):
 InternalCompilerError: Assembly exception for bytecode


Error HH600: Compilation failed
```